File: 백준/Silver/10799. 쇠막대기/쇠막대기.swift
let input = Array(readLine()!)

var laserCnt = 0
var validNum = 0
var ans = 0

for i in 0..<input.count {
    if input[i] == "(" {
        validNum += 1
    } else if input[i] == ")" {
        validNum -= 1
        if input[i-1] == "(" {
            ans += validNum
        } else {
            ans += 1
        }
    }
}
print(ans)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int ans = 0;
        int validNum = 0;

        char[] stick = br.readLine().toCharArray();

        for(int i =0; i< stick.length; i++) {
            if( stick[i] == '(') {
                validNum++;
            } else if (stick[i] == ')') {
                validNum--;
                if(stick[i-1] == '(') {
                    ans += validNum;
                } else {
                    ans += 1;
                }
            }

        }
        System.out.println(ans);
    }
}
